microsoft/testsuites/stress/stress_ng_suite.py

```diff
@@ -0,0 +1,161 @@
+from typing import Any, Dict, List, cast
+
+from lisa import (
+    Environment,
+    RemoteNode,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    notifier,
+)
+from lisa.messages import SubTestMessage, TestStatus, create_test_result_message
+from lisa.testsuite import TestResult
+from lisa.tools import StressNg
+from lisa.util import SkippedException
+from lisa.util.process import Process
+
+
+@TestSuiteMetadata(
+    area="stress",
+    category="stress-ng",
+    description="""
+    A suite for running the various classes of stressors provided
+    by stress-ng.
+    """,
+)
+class StressNgTestSuite(TestSuite):
+    TIME_OUT = 3600
+    CONFIG_VARIABLE = "stress_ng_jobs"
+
+    @TestCaseMetadata(
+        description="""
+        Runs a stress-ng jobfile. The path to the jobfile must be specified using a
+        runbook variable named "stress_ng_jobs". For more info about jobfiles refer:
+        https://manpages.ubuntu.com/manpages/jammy/man1/stress-ng.1.html
+        """,
+        priority=4,
+    )
+    def verify_stress_ng_jobfile(
+        self,
+        variables: Dict[str, Any],
+        environment: Environment,
+        result: TestResult,
+    ) -> None:
+        if self.CONFIG_VARIABLE in variables:
+            jobs = variables[self.CONFIG_VARIABLE]
+            for job_file in jobs:
+                self._run_stress_ng_job(job_file, environment, result)
+        else:
+            raise SkippedException("No jobfile provided for stress-ng")
+
+    @TestCaseMetadata(
+        description="Runs stress-ng's 'cpu' class stressors for 60s each.",
+        priority=4,
+    )
+    def verify_stress_ng_cpu_stressors(
+        self,
+        environment: Environment,
+    ) -> None:
+        self._run_stressor_class(environment, "cpu")
+
+    @TestCaseMetadata(
+        description="Runs stress-ng's 'memory' class stressors for 60s each.",
+        priority=4,
+    )
+    def verify_stress_ng_memory_stressors(
+        self,
+        environment: Environment,
+    ) -> None:
+        self._run_stressor_class(environment, "memory")
+
+    @TestCaseMetadata(
+        description="Runs stress-ng's 'vm' class stressors for 60s each.",
+        priority=4,
+    )
+    def verify_stress_ng_vm_stressors(
+        self,
+        environment: Environment,
+    ) -> None:
+        self._run_stressor_class(environment, "vm")
+
+    @TestCaseMetadata(
+        description="Runs stress-ng's 'io' class stressors for 60s each.",
+        priority=4,
+    )
+    def verify_stress_ng_io_stressors(
+        self,
+        environment: Environment,
+    ) -> None:
+        self._run_stressor_class(environment, "io")
+
+    @TestCaseMetadata(
+        description="Runs stress-ng's 'network' class stressors for 60s each.",
+        priority=4,
+    )
+    def verify_stress_ng_network_stressors(
+        self,
+        environment: Environment,
+    ) -> None:
+        self._run_stressor_class(environment, "network")
```

Comment:
Those test cases start process, but no verification, no data collected. What's they for?

Reply:
There is return code verification for the process. For now, we just want to see if any kernel panic (or something as bad) happens while stressing. Panic will cause the test to fail even without any explicit verification. In the future we will add more dmesg log validations.

Reply:
If so, use the serial console to check the kernel panic. That would give more detailed error information.

Reply:
Sure, we will use the serial console for verifying the logs. But this will be a future improvement. In libvirt platform serial console logs are captured automatically for all nodes. For now, these can be checked manually while investigating kernel panics.

Reply:
"Manually" is not a scalable way. Check panic is simple, refer to below example. If there are more patterns in libvirt, they can be added in future.

https://github.com/squirrelsc/LISA/blob/extension_prototype/microsoft/testsuites/core/provisioning.py#L258

Reply:
Added `check_panic`.